app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/NewPasswordScreen.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.simply.birthdayapp.R
+import com.simply.birthdayapp.presentation.ui.components.AppBaseTopBar
+import com.simply.birthdayapp.presentation.ui.components.AuthButton
+import com.simply.birthdayapp.presentation.ui.components.PasswordTextFiled
+import com.simply.birthdayapp.presentation.ui.theme.AppTheme
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun NewPasswordScreen(
+    forgotPasswordViewModel: ForgotPasswordViewModel,
+    onResetPasswordSuccess: () -> Unit = {},
+    onCodeInvalid: () -> Unit = {},
+    navToFprgotPasswordScreen: () -> Unit = {}
+) {
+    val password by forgotPasswordViewModel.password.collectAsState()
+    val hasPasswordError by forgotPasswordViewModel.hasPasswordError.collectAsState()
+    val hasRepeatPasswordError by forgotPasswordViewModel.hasRepeatPasswordError.collectAsState()
+    val repeatPassword by forgotPasswordViewModel.repeatPassword.collectAsState()
+    val doneButtonEnabled by forgotPasswordViewModel.enableDoneButton.collectAsState()
+    val resetPasswordSuccess by forgotPasswordViewModel.resetPasswordSuccess.collectAsState()
+    val resetPasswordErrorState by forgotPasswordViewModel.resetPasswordErrorState.collectAsState()
+    val resetPasswordErrorMessage by forgotPasswordViewModel.resetPasswordErrorMessage.collectAsState()
+    val context = LocalContext.current
+
+    LaunchedEffect(resetPasswordSuccess) {
+        if (resetPasswordSuccess) {
+            Toast.makeText(context, R.string.reset_passwors_success, Toast.LENGTH_SHORT).show()
+            onResetPasswordSuccess()
+            forgotPasswordViewModel.resetSuccessState()
+        }
+    }
+    LaunchedEffect(resetPasswordErrorMessage) {
+        if (resetPasswordErrorMessage.isNotEmpty()) {
+            Toast.makeText(context, R.string.invalid_code, Toast.LENGTH_SHORT).show()
+            onCodeInvalid()
+            forgotPasswordViewModel.resetRegisterErrorMessage()
+        }
+    }
+
+    if (resetPasswordErrorState) {
+        AlertDialog(
+            containerColor = AppTheme.colors.white,
+            titleContentColor = AppTheme.colors.black,
+            textContentColor = AppTheme.colors.black,
+            onDismissRequest = { forgotPasswordViewModel.resetPasswordErrorState() },
+            title = { Text(text = stringResource(id = R.string.reset_password_error)) },
+            text = { Text(text = stringResource(R.string.reser_password_network_error)) },
+            confirmButton = {
+                TextButton(onClick = { forgotPasswordViewModel.resetPasswordErrorState() }) {
+                    Text(text = stringResource(id = R.string.ok), color = AppTheme.colors.black)
+                }
+            },
+        )
+    }
+    Scaffold(
```

Comment:
Nesting scaffolds is not a good practice. It may lead to crashes in the future. You can wrap your appBar with a column if you need it on top.